### PR TITLE
feat(volume group): add rest core gRPC implementation

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**

### CODE REVIEW

File: control-plane/grpc/proto/v1/volume/volume.proto [hunk 1]
```diff
@@ -170,6 +178,8 @@ message CreateVolumeRequest {
   optional Topology topology = 7;
   // flag indicating thin provisioning
   bool thin = 8;
+  // Volume Group related information.
+  optional VolumeGroup volume_group = 10;
```
Comment (tiagolobocastro) [inline control-plane/grpc/proto/v1/volume/volume.proto]: should be 9

File: control-plane/grpc/src/operations/volume/traits.rs [hunk 2]
```diff
@@ -115,6 +115,7 @@ impl From<VolumeSpec> for volume::VolumeDefinition {
                 topology: volume_spec.topology.map(|topology| topology.into()),
                 last_nexus_id: nexus_id.map(|id| id.to_string()),
                 thin: volume_spec.thin,
+                volume_group: volume_spec.volume_group.map(|vg| vg.into()),
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/volume/traits.rs]: nit: might be able to use into_opt, but both are fine.

File: control-plane/grpc/src/operations/volume/traits.rs [hunk 3]
```diff
@@ -674,6 +676,8 @@ pub trait CreateVolumeInfo: Send + Sync + std::fmt::Debug {
     fn labels(&self) -> Option<VolumeLabels>;
     /// Flag indicating whether the volume should be thin provisioned
     fn thin(&self) -> bool;
+    /// Volume Group related information.
+    fn volume_group(&self) -> Option<VolumeGroup>;
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/volume/traits.rs:680]: non-related: I would like to do without all these traits at some point and use simple types with the standard From et-all traits.

File: control-plane/stor-port/src/types/v0/store/volume.rs [hunk 4]
```diff
@@ -166,6 +166,9 @@ pub struct VolumeSpec {
     /// The publish context of the volume.
     #[serde(default)]
     pub publish_context: Option<HashMap<String, String>>,
+    #[serde(default)]
+    /// Volume Group related information.
```
Comment (tiagolobocastro) [inline control-plane/stor-port/src/types/v0/store/volume.rs]:
```suggestion
    /// Volume Group related information.
    #[serde(default)]
```
simply for consistency

File: control-plane/grpc/src/operations/volume/traits.rs (hunk 3 above)
Comment (tiagolobocastro) [reply control-plane/grpc/src/operations/volume/traits.rs:680]: And do without most of the transport etc one day :)

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/grpc/src/operations/volume/traits.rs (hunk 3 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/operations/volume/traits.rs:680]: Yeah, that would be really good. :-)

File: control-plane/stor-port/src/types/v0/store/volume.rs (hunk 4 above)
Comment (Abhinandan-Purkait) [reply control-plane/stor-port/src/types/v0/store/volume.rs]: Right.

File: control-plane/grpc/proto/v1/volume/volume.proto (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/proto/v1/volume/volume.proto]: Oops, right.

File: control-plane/grpc/src/operations/volume/traits.rs (hunk 2 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/operations/volume/traits.rs]: Done

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**